src/components/SignUpForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Formik, Field, Form, FormikHelpers } from 'formik';
+import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
+import { useNavigate, Link } from "react-router-dom";
+import hiddenInput from '../assets/icons/hidden-input.png';
+import './SignUpForm.css';
+
+interface Values {
+    email: string;
+    password: string;
+    confirmedPassword: string;
+}
+
+function SignUpForm() {
+    const auth = getAuth();
+    const navigate = useNavigate();
+    const [errorInfo, setErrorInfo] = useState("");
+
+    const checkIfSamePassword = (password: string, confirmedPassword: string): Boolean => {
+        if (password === confirmedPassword) {
+            return true;
+        } else {
+            setErrorInfo("Error: Passwords differ. Please try again.");
+            return false;
+        }
+    }
+
+    const handleSubmit = (
+        values: Values,
+        { setSubmitting }: FormikHelpers<Values>
+    ) => {
+        const { email, password, confirmedPassword } = values;
+
+        if (password) {
+            const passwordIsSame = checkIfSamePassword(password, confirmedPassword);
+            if (passwordIsSame) {
+                setSubmitting(false);
+                createUserWithEmailAndPassword(auth, email, password)
+                    .then((response) => {
+                        navigate("/");
+                        setErrorInfo("");
+                    })
+                    .catch((error) => {
+                        const errorCode = error.code;
+                        setErrorInfo(`Error: ${errorCode}. Please try again.`);
+                    });
```

Comment:
Use redux-saga for all async calls
And lets try to use async/await instead of Promise